template/src/index.ts.ejs

```diff
@@ -50,6 +50,8 @@ export function <%- ex.name %>(): number {
       <% } -%>
     <% } else if (ex.output.type === 'string') { -%>
       Host.outputString(output)
+    <% } else if (ex.output.type === 'number') { -%>
+      Host.outputBytes(Float32Array.of(output).buffer)
```

Comment:
Thanks for the contribution! I think there is a misunderstanding of what this is doing though and this isn't something we generally support. At the boundary level (inputs and outputs), every piece of data that goes across needs an encoding. we can't really pass bytes across purely based on types. it might seem like a contradiction given the case above checks for type === 'string', but that's a backward compat thing and will eventually be changed to something like `isUtf8Encoded(ex.output)`.

So to pass a float you need to do here is either choose an existing encoding or we create a new encoding. Two possible encoding you could choose would be either json or raw binary.

example w/ contentType `application/json`:
```
// note: precision is not really limited by the JSON format but rather by the language
> JSON.stringify(3.1415926535)
'3.1415926535'
> JSON.parse('3.1415926535')
3.1415926535
```

example w/ contentType = `application/x-binary`

```
// note: your precision can be 32 or 64 bit depending on your need
> new Uint8Array((new Float32Array([3.145926])).buffer)
Uint8Array(4) [ 218, 86, 73, 64 ]
> new Float32Array(z.buffer)[0];
3.1459264755249023
```

Reply:
Thanks for the detailed explanation! So, just to be clear, the way my schema is currently set up, the default plugin constructed from the bundle throws an error. If I am using `application/x-binary`  with `number` an `float` as my `type` and `format`, then I will be expected to implement that interface myself. If I am using `application/json` or another potential standard encoding, then that template should be expected to work out of the box. Is my understanding accurate?